Sources/SwiftParser/Parameters.swift

```diff
@@ -156,7 +156,19 @@ extension Parser {
       defaultValue = nil
     }
 
-    let trailingComma = self.consume(if: .comma)
+    var trailingComma: Token?
+    if self.at(.comma) {
+      trailingComma = self.consume(if: .comma)
+    } else if !self.at(.rightParen) {
+      let canParseIdentifier: Bool = withLookahead {
+        $0.canParseTypeIdentifier(allowKeyword: false)
+          || ($0.consume(if: .atSign) != nil && $0.canParseCustomAttribute())
```

Comment:
Sorry if I wasn’t clear in my previous message. I meant that we need separate `withLookahead` calls to avoid the following: `canParseTypeIdentifier` consumes some tokens but then returns `false`. We thus evaluate the rhs of `||`, which might return `true` only because we previously consumed tokens. Looking at the code, I think we would hit this with the following:

```
func foo(x: Int Int<@abc)
```

`Int<` will be consumed by `canParseTypeIdentifier` and then we end up in `($0.consume(if: .atSign) != nil && $0.canParseCustomAttribute())`